core/services/pipeline/runner_test.go

```diff
@@ -332,12 +324,41 @@ answer1 [type=median                      index=0];
 	_, trrs, err := r.ExecuteRun(ctx, spec, nil, pipeline.JSONSerializable{}, *logger.Default)
 	require.NoError(t, err)
 	for _, trr := range trrs {
-		if trr.IsTerminal {
+		if trr.IsTerminal() {
 			require.Equal(t, decimal.RequireFromString("1100"), trr.Result.Value.(decimal.Decimal))
 		}
 	}
 }
 
+func Test_PipelineRunner_MultipleOutputs(t *testing.T) {
+	store, cleanup := cltest.NewStore(t)
+	defer cleanup()
+	orm := new(mocks.ORM)
+	orm.On("DB").Return(store.DB)
+	r := pipeline.NewRunner(orm, store.Config)
+	input := map[string]interface{}{"val": 2}
+	_, trrs, err := r.ExecuteRun(context.Background(), pipeline.Spec{
+		DotDagSource: `
+a [type=multiply input="$(input.val)" times=2]
+b1 [type=multiply input="$(a)" times=2]
+b2 [type=multiply input="$(a)" times=3]
+c [type=median values=<[ $(b1), $(b2) ]> index=0]
+a->b1->c;
+a->b2->c;`,
+	}, input, pipeline.JSONSerializable{}, *logger.Default)
+	require.NoError(t, err)
+	require.Equal(t, 4, len(trrs))
+	assert.Equal(t, false, trrs.FinalResult().HasErrors())
+
+	// a = 4
+	// (b1 = 8) + (b2 = 12)
+	// c = 20 / 2
+
+	result, err := trrs.FinalResult().SingularResult()
```

Comment:
The test is labelled MultipleOutputs, but there is only one output?

Reply:
I can add a test for multiple final outputs too (I assumed that already worked via that `Index/OutputIndex` param), this test confirms that a diamond shaped graph works:

```
a -> b1 -> c
a -> b2 -> c
```
Where `a` has multiple outputs. This was not possible before because each task had a single `OutputTask` (and also because there was one goroutine for each root, in this case one, but the next step would have required two of them).

Reply:
I was imagining a task that terminates with multiple outputs. i.e. there is no one single output task